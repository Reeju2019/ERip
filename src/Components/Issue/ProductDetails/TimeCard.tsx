import * as React from 'react'

interface ITimeCardProps {
  time: string
  setActiveTime: React.Dispatch<React.SetStateAction<string>>
}

const TimeCard: React.FunctionComponent<ITimeCardProps> = ({ time, setActiveTime }) => {
  const TimeClicked = () => {
    setActiveTime(time)
  }
  return (
    <>
      <div onClick={TimeClicked}>{time}</div>
    </>
  )
}

export default TimeCard
