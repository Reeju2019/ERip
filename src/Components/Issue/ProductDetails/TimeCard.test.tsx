import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'
import ReactDOM from 'react-dom'
import TimeCard from './TimeCard'


const setActiveTime = jest.fn()

const time = "a"
test('time card', () => {
    render(<TimeCard time={time} setActiveTime={setActiveTime} />)
})
describe('time card rendering specification', () => {
    it('SnapShot testing', () => {
        const component = renderer.create(<TimeCard time={time} setActiveTime={setActiveTime} />)
        const tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
    it('render without crashing time card', () => {
        const div = document.createElement('div')
        render(<TimeCard time={time} setActiveTime={setActiveTime} />)
        ReactDOM.unmountComponentAtNode(div)
    })
})
