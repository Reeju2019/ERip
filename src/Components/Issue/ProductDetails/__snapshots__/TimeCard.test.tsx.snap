// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`time card rendering specification SnapShot testing 1`] = `
<div
  onClick={[Function]}
>
  a
</div>
`;
